components/assets/AssetActionButtons/RedeemButton.tsx

```diff
@@ -119,6 +120,11 @@ export const RedeemButtonByValue = ({
   const wallet = useWallet();
   const signer = wallet?.getActiveSigner();
   const notificationStore = useNotifications();
+  const router = useRouter();
+  const walletAddress = wallet?.selectedAddress;
+  const addressFromUrl = Array.isArray(router.query.address)
```

Comment:
Not sure if this is the best approach, won't this cause the button to be disabled if it is used on a page that doesn't have the user's address in the route? I think it would be better to do a balance check for the connected wallet